Make make_tensor in broadcast utilities public and rename free_broadcast_tensor

tarun292 opened the pull request:
Summary:
This diff does a couple of things:
- Makes `make_tensor` a public function so that we can create temporary intermediate tensors in operators that need to do so. (Such as NMS that is implemented above in this stack)
- Renames `free_broadcast_tensor` to a more generic name `free_tensor`

Differential Revision: D55577026

iseeyuan commented on kernels/portable/cpu/util/broadcast_util.cpp:
```diff
@@ -19,16 +19,14 @@ namespace executor {
 }
 
-namespace {
-
```
It's interesting that malloc is used in this function but I don't see how the malloced pointer is freed. @SS-JIA @manuelcandales , could you provide more context that I miss? cc @dbort

tarun292 replied:
There's a corresponding function called `free_tensor` that when passed in the tensor created via this function free's it up.

manuelcandales replied:
@tarun292 These functions are not being used in the portable kernels. I believe you introduced these utils back in January 2023, and maybe they were being used at that time, but not anymore. They should be moved out of the portable lib utils.

tarun292 committed:
Add sort util for 1D tensors

Differential Revision: D55577025

tarun292 committed:
Make make_tensor in broadcast utilities public and rename free_broadcast_tensor (#2785)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/2785

This diff does a couple of things:
- Makes `make_tensor` a public function so that we can create temporary intermediate tensors in operators that need to do so. (Such as NMS that is implemented above in this stack)
- Renames `free_broadcast_tensor` to a more generic name `free_tensor`

Differential Revision: D55577026
